abn/service_impl.go

```diff
@@ -47,6 +47,9 @@ func lookupInternal(application string, user string) (controllers.RoutemapInterf
 	}
 
 	// record user; ignore error if any; this is best effort
+	if MetricsClient == nil {
+		return nil, invalidVersion, fmt.Errorf("no metrics client")
+	}
```

Comment:
This check exists in a couple places. Originally, I added this because I wasn't sure what the direction we were moving towards. Seems to me that we should consider having some validation somewhere so we do not need to have this check everywhere.

Additionally, the `MetricsClient` is also referenced in `server.go` and possible other places. I wonder if this cluster local variable should be moved somewhere else.